syntax = "proto3";

package RankProtocol;

message RankItem {
    uint64 role_id = 1;
    int32 rank = 2;
    int32 score = 3;
    string name = 4;
    string avatarurl = 5;
}

message C2S_RankLoad {
    int32 subtype = 1;
}

message S2C_RankLoad {
    int32 ret = 1;
    int32 subtype = 2;
    RankItem myself = 3;
    repeated RankItem ris = 4;
}

message C2S_RankUpdate {
    int32 subtype = 1;
    int32 score = 2;
}

message S2C_RankUpdate {
    int32 ret = 1;
}

message C2S_RankMyself {
    int32 subtype = 1;
}

message S2C_RankMyself {
    int32 ret = 1;
    int32 subtype = 2;
    RankItem myself = 3;
}

message S2C_RankNotify {
    int32 gameid = 1;
    int32 subtype = 2;
    repeated RankItem ris = 3;
}
